src/sipsess/reply.c

```diff
@@ -205,16 +217,33 @@ int sipsess_reply_1xx(struct sipsess *sess, const struct sip_msg *msg,
 				  421, "Extension required",
 				  "Require: 100rel\r\n"
 				  "Content-Length: 0\r\n\r\n");
-		return -1;
+		return EPROTO;
 	}
 	else if (rel100_peer == REL100_REQUIRED && !rel100) {
 		(void)sip_treplyf(&sess->st, NULL, sess->sip, msg, false, 420,
 				  "Bad Extension", "Unsupported: 100rel\r\n"
 				  "Content-Length: 0\r\n\r\n");
-		return -1;
+		return EPROTO;
 	}
 
 	reliably = rel100 && rel100_peer && scode != 100;
+
+	if (reliably && sess->neg_state == SDP_NEG_NONE
+	    && !mbuf_get_left(desc))
+		return EINVAL;
+
+	if (sess->neg_state == SDP_NEG_NONE) {
+		if (reliably && !mbuf_get_left(desc))
+			return EINVAL;
+		else if (!reliably && mbuf_get_left(desc))
+			desc = NULL;
+	}
+	else if ((sess->neg_state == SDP_NEG_DONE
+		 || sess->neg_state == SDP_NEG_LOCAL_OFFER)
+		 && mbuf_get_left(desc)) {
+		desc = NULL;
```

Comment:
Do we need the check `mbuf_get_left(desc)` here?

If it is zero we could also set it to NULL, right?

Reply:
Good point, this check is unnecessary.  I removed it here and in the other cases where we set `desc = NULL`.